### PR TITLE
Activity log for new case edit fields

### PR DESCRIPTION
**Pull request by KyleMacPherson**
https://trello.com/c/1mTckxw5/1563-add-case-name-and-case-reference-changes-to-case-activity-log
https://trello.com/c/re7eY1YM/1564-send-email-notification-when-case-name-is-updated

## Description
This PR ecompasses two tickets from our trello board.
1. Add changes to the activity log when a user edits case name or reference number.
2. Send users a notification email when the case name of a case they own is edited by someone else.

## Checklist:
- [ ] Have you documented your changes in the pull request description?
- [ ] Does the change present any security considerations?
- [ ] Is any gem functionality overloaded? Eg: Devise controller methods being overloaded.
- [ ] Has acceptance criteria been tested by a peer?
- [ ] Has the CHANGELOG been updated? (If change is worth telling users about.)

### General testing (author)
- [ ] Test without JavaScript
- [ ] Test on small screen

### Accessibility testing (author)
- [ ] Works keyboard only
- [ ] Tested with one screen reader
- [ ] Zoom page to 400% - content still visible
- [ ] Disable CSS - does content make sense and appear in a logical order?

### CODE REVIEW

**Review by nasirkhanbeis — APPROVED** [review]
Tested this on the review app, looks good as expected.

**Review by stdstaff — APPROVED** [review]
Couple of newlines at the ends of files to put back...

File: app/services/change_case_name.rb [hunk 1]
```diff
@@ -0,0 +1,57 @@
+class ChangeCaseName
+  include Interactor
+  include EntitiesToNotify
+
+  delegate :investigation, :user_title, :user, to: :context
+
+  def call
+    context.fail!(error: "No investigation supplied") unless investigation.is_a?(Investigation)
+    context.fail!(error: "No reference number supplied") unless user_title.is_a?(String)
```
Comment (slorek) [inline app/services/change_case_name.rb:9]:
```suggestion
    context.fail!(error: "No case name supplied") unless user_title.is_a?(String)
```

File: spec/features/edit_investigation_case_name_spec.rb [hunk 2]
```diff
@@ -1,7 +1,8 @@
 require "rails_helper"
 
-RSpec.feature "Edit an investigation's case name", :with_stubbed_opensearch, :with_stubbed_antivirus, :with_stubbed_mailer do
+RSpec.feature "Edit an investigation's case name", :with_opensearch, :with_stubbed_mailer do
```
Comment (slorek) [inline spec/features/edit_investigation_case_name_spec.rb:3]: Not sure this change is needed; it'll make the spec run a lot slower.

File: spec/features/edit_investigation_reference_number_spec.rb [hunk 3]
```diff
@@ -1,13 +1,14 @@
 require "rails_helper"
 
-RSpec.feature "Edit an investigation's reference number", :with_stubbed_opensearch, :with_stubbed_antivirus, :with_stubbed_mailer do
+RSpec.feature "Edit an investigation's reference number", :with_opensearch, :with_stubbed_mailer, type: :feature do
```
Comment (slorek) [inline spec/features/edit_investigation_reference_number_spec.rb:3]: Not sure this change is needed; it'll make the spec run a lot slower.

**Review by slorek — APPROVED** [review]

**Pull request merged**